### ISSUE TITLE
undefined method `to_ary' for #<Grape::Router::Route:0x007fc379d7e490> in route.rb

### ISSUE THREAD

**Issue by atomless** [ruby-grape/grape#1450]
I'm trying to get to the bottom of a long delay in my ruby server starting up that has only recently been introduced by some change or other and when running the server with warnings and debugging to verbose I'm curious why I see pages and pages of this error outputted on server start

```
Exception `NoMethodError' at ~/.rvm/gems/ruby-2.1.9/gems/grape-0.16.2/lib/grape/router/route.rb:27 - undefined method `to_ary' for #<Grape::Router::Route:0x007fc379d7e490>
```

I looked in route.rb and there is no use of to_ary!

**Comment by dblock** [comment]
Interesting. I would monkey-patch that code and see what's being called and why.

**Comment by namusyaka** [comment]
Yeah I'd like to see repo for reproducing the issue

**Comment by 0x01f7** [comment]
solution:

```ruby
class Grape::Router::Route
  private 
  def to_ary
    nil
  end
end
```

ref: 

* [Implement `to_ary` to avoid calls to method_missing](https://github.com/bundler/bundler/pull/1274)
* [TIL: It's OK to return nil from to_ary](https://tenderlovemaking.com/2011/06/28/til-its-ok-to-return-nil-from-to_ary.html)

**Comment by dblock** [comment]
But who's calling this @0x01f7? Why is this being called at all?

**Comment by 0x01f7** [comment]
Gemfile:

```ruby
ruby '2.3.1'

gem 'grape', '0.17.0'
gem 'grape-swagger', '0.23.0'
```

----

config.ru:

```ruby
require 'grape'
require 'grape-swagger'

module Demo
  class API < ::Grape::API
    add_swagger_documentation
  end
end

run Demo::API
```

----

command line:

```shell
$ bundle exec rackup -d
```

----

backtrace:

```text
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/endpoint.rb:168:in `flatten'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/endpoint.rb:168:in `to_routes'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/endpoint.rb:132:in `routes'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/endpoint.rb:132:in `collect'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/endpoint.rb:132:in `routes'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/api.rb:66:in `map'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/api.rb:66:in `prepare_routes'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-0.17.0/lib/grape/dsl/routing.rb:181:in `routes'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-swagger-0.23.0/lib/grape-swagger.rb:60:in `combine_routes'
~/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/grape-swagger-0.23.0/lib/grape-swagger.rb:39:in `add_swagger_documentation'
```

**Comment by dblock** [comment]
Can you try to turn this into a spec in Grape, without grape-swagger? Assuming that it's not monkey-patching anything, we have code that is available but not called in Grape proper via `routes`.

**Comment by 0x01f7** [comment]
* It has been fixed in v0.19.0 by introducing `Grape::Router::Route#respond_to_missing?`.
* Use `rubocop`, and enable [Style/MethodMissing](https://github.com/bbatsov/ruby-style-guide#no-method-missing) to avoid it.

**Comment by dblock** [comment]
Thanks @0x01f7. Closing. Reopen if you think there's something else to do. If we're not catching this with Rubocop please PR.
